From 5b1c033918dbea39c3e9ce5d5e36e1bebec2c4ae Mon Sep 17 00:00:00 2001
From: Dave Reisner <dreisner@archlinux.org>
Date: Thu, 27 Sep 2012 20:48:13 -0400
Subject: [PATCH] tmpfiles: restore previous behavior for F/f

d4e9eb91ea changed the behavior for the F and f actions, wrongly sending
them to glob_item(). Restore the old behavior and shortcut straight to
write_one_file().
(cherry picked from commit 1845fdd967d3a4c06f895413505de3c2429955b0)
---
 src/tmpfiles/tmpfiles.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/tmpfiles/tmpfiles.c b/src/tmpfiles/tmpfiles.c
index 30b0b93..e901ec4 100644
--- a/src/tmpfiles/tmpfiles.c
+++ b/src/tmpfiles/tmpfiles.c
@@ -652,6 +652,10 @@ static int create_item(Item *i) {
 
         case CREATE_FILE:
         case TRUNCATE_FILE:
+                r = write_one_file(i, i->path);
+                if (r < 0)
+                        return r;
+                break;
         case WRITE_FILE:
                 r = glob_item(i, write_one_file);
                 if (r < 0)
